azkaban-common/src/main/java/azkaban/event/EventData.java

```diff
@@ -0,0 +1,34 @@
+package azkaban.event;
+
+import azkaban.executor.Status;
+
+/**
+ * Carries a snapshot of the status data, suitable for asynchronous message passing.
+ */
+public class EventData {
+
+  private Status status;
+  private String nestedId;
+
+  public static EventData create(Status status) {
```

Comment:
I mimicked the style of existing code like `Event.create`. As far as I'm concerned, it could be a public constructor, too.